### PR TITLE
added documentation to help other developers on this project

### PR DESCRIPTION
**Pull request by jscancella**
I discovered a couple of things while working on a different pull request for this project that I thought would help the next developer, so I added it to the README in a new section on development.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -114,6 +114,34 @@ dependencies {
 }
 ```
 
+## Development
+### Setup
+* development requires java 11 to be installed
+* development assumes you are running on Ubuntu, while you may be able to use other operating systems they are not supported for developement.
+* before creating commits
+  * read https://www.conventionalcommits.org/en
+  * create the following script in your .git/hooks directory and name it commit.msg
```
Comment (KengoTODA) [inline README.md:123]: if we really need a script to help using conventional commit, we can introduce [Commitizen](https://commitizen.github.io/cz-cli/) for better automation.

File: README.md [hunk 2]
```diff
@@ -114,6 +114,34 @@ dependencies {
 }
 ```
 
+## Development
+### Setup
+* development requires java 11 to be installed
```
Comment (KengoTODA) [inline README.md:119]: We can use Java 11 or later, upper versions like 14 should work.

File: README.md [hunk 3]
```diff
@@ -114,6 +114,34 @@ dependencies {
...
+* development assumes you are running on Ubuntu, while you may be able to use other operating systems they are not supported for developement.
```
Comment (KengoTODA) [inline README.md:120]: Do we need this sentence? I'm developing on macOS, and it seems working as expected.
If we know any trouble with Windows or other OS, I want to fix it.

File: README.md (hunk 1 above)
Comment (jscancella) [reply README.md:123]: Ok. I based it on the fact that I ran into errors when doing the functional testing in my Windows machine and that we use the Ubuntu image on the CI server. When I get a chance I will update it.

File: README.md (hunk 3 above)
Comment (jscancella) [reply README.md:120]: Ok. I based it on the fact that I ran into errors when doing the functional testing in my Windows machine and that we use the Ubuntu image on the CI server. When I get a chance I will update it.

File: README.md [hunk 4]
```diff
@@ -114,6 +114,34 @@ dependencies {
...
 
+## Development
+### Setup
+* development requires java 11 or higher to be installed
+* The CI server uses `ubuntu-latest` docker image, but you should be able to develop on any linux/unix based OS.
+* before creating commits
+  * read https://www.conventionalcommits.org/en
+  * Optionally create the following script in your .git/hooks directory and name it commit.msg. This will ensure that your commits follow the covential commits pattern.
+```python
+  #!/usr/bin/env python
```
Comment (KengoTODA) [inline README.md:125]:
```suggestion
#!/usr/bin/env python
```

File: README.md [hunk 5]
```diff
@@ -115,6 +115,34 @@ dependencies {
...
+    pattern = r'(build|ci|docs|feat|fix|perf|refactor|style|test|chore|revert)(\([\w\-]+\))?:\s.*'
+    filename = sys.argv[1]
+    ss = open(filename, 'r').read()
+    m = re.match(pattern, ss)
+    if m == None: raise Exception("Conventional commit validation failed. Did you forget to add one of the allowed prefixes? (build|ci|docs|feat|fix|perf|refactor|style|test|chore|revert)")
+
+if __name__ == "__main__":
+    main()
+  ```
+* when running gradle, do so using the `.gradlew` script in this directory
```
Comment (KengoTODA) [inline README.md:143]:
```suggestion
* when running gradle, do so using the `gradlew` script in this directory
```

**Pull request merged**